ModularTegustation/tegu_mobs/apocalypse_bird.dm

```diff
@@ -152,13 +180,106 @@
 	visible_message("<span class='danger'>[src] suddenly disappears!</span>")
 	forceMove(T)
 	visible_message("<span class='danger'>[src] suddenly appears in front of you!</span>")
-	SLEEP_CHECK_DEATH(3 SECONDS)
+	SLEEP_CHECK_DEATH(2 SECONDS)
+	teleport_cooldown = world.time + teleport_cooldown_time
+	attacking = FALSE
+
+/mob/living/simple_animal/hostile/megafauna/apocalypse_bird/proc/Bite()
+	if(attacking || !bite_possible)
+		return
+	attacking = TRUE
+	var/list/candidates = list()
+	for(var/mob/living/L in range(16, src))
+		if(faction_check_mob(L, FALSE))
+			continue
+		if(L.stat == DEAD)
+			continue
+		candidates += L
+	var/dir_to_target = pick(GLOB.cardinals)
+	if(LAZYLEN(candidates))
+		dir_to_target = get_cardinal_dir(get_turf(src), get_turf(pick(candidates)))
+	var/turf/source_turf = get_turf(src)
+	var/turf/area_of_effect = list()
+	var/turf/upper_bound
+	var/turf/lower_bound
+	switch(dir_to_target)
```

Comment:
Can't you just use getline for this, it would work nicely.

Reply:
for the WIDTH of it, you could probably make it a radius around... like a Hybrid between Woodsman's AoE and Hello.
So if it was 5 wide you'd do a radius of 2 and step do a loop of `for(var/k = 2 to bite_length-2)` to "step" there